internal/introspection/idlfile.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package introspection
+
+import "strings"
+
+// IDLFile is a generic IDL file. For example, a thrift or protobuf IDL.
+type IDLFile struct {
+	// FilePath starts from an arbitrary root containing the file and all its
+	// includes.
+	FilePath string `json:"filePath"`
+
+	// Checksum of Content. The format is: "${type}:${hex}". For example, a
+	// sha1 checksum would look like:
+	// "sha1:d55f7ce6138889e9e45b62df3741820777c65fcd".
+	Checksum string `json:"checksum"`
```

Comment:
I hear you.  Thing is, I would prefer to treat the checksum as simply as possible. I could have used a slice of bytes. But I have decided to go with the convention of a printable string. With this simple string, the json output is simple. I really do not thing there is any need to complicate this here. 

At first i had named the checksum SHA1. Which basically made yarpc forcing you to implement a specific hash type. In exchange consumer of yarpc know they have a single checksum/hash to deal with. With an enum, yarpc would now force you to deal with different checksum types. Worst, you can only emit checksum types vetted by yarpc now. With a hash type as a string (here, part of the string because why not), yarpc is removed from any decision on the matter.

So, as I see it, we have 3 choices:
 1. we pick a single checksum types, and stick with it.
 2. we offer a fixed list of checksum types.
 3. the checksum is a self-contained string.

I really want to start simple, and I am in favor of option 3.